src/types/yara_rule.cpp

```diff
@@ -57,9 +41,9 @@ const YaraMeta* YaraRule::getMeta(std::string id) const
  * @param index Index of selected match (indexed from 0)
  * @return Pointer to selected match or @c nullptr if such match is not found
  */
-const YaraMatch* YaraRule::getMatch(std::size_t index) const
+const YaraMatch* YaraRule::getMatch(std::size_t index = {}) const
```

Comment:
If you want to add default parameter to this method do so in header file. This way, only this file would be able to see this default parameter but you would not be able to use it outside this compile unit. Also reconsider whether this change is really needed. There is already method `getFirstMatch()` so I don't really see this change to be that useful.